torch/csrc/jit/operator.cpp

```diff
@@ -203,49 +207,179 @@ struct SchemaParser {
   }
   Lexer L;
   bool kwarg_only;
-  at::Tensor one = at::full({}, 1, at::kLong);
-  at::Tensor zero = at::full({}, 0, at::kLong);
+  static at::Tensor one() {
+    static at::Tensor v = at::full({}, 1, at::kLong);
+    return v;
+  }
+  static at::Tensor zero() {
+    static at::Tensor v = at::full({}, 0, at::kLong);
+    return v;
+  }
 };
 }
 
-using SchemaMap = std::unordered_map<std::string, std::vector<FunctionSchema>>;
 
-// defined in aten_schema_declarations.cpp
-extern const char * schema_declarations;
+namespace {
+
+using OperatorMap = std::unordered_map<Symbol, std::vector<std::shared_ptr<Operator>>>;
+struct OperatorRegistry  {
+  OperatorMap operators;
+  std::mutex lock;
+  void registerOperator(Operator&& op){
+    std::lock_guard<std::mutex> guard(lock);
+    Symbol sym = Symbol::fromQualString(op.schema.name);
+    auto it = operators.find(sym);
+    if(it == operators.end()) {
+      it = operators.insert({sym, {}}).first;
+    }
+    it->second.push_back(std::make_shared<Operator>(std::move(op)));
+  }
+  const std::vector<std::shared_ptr<Operator>>& getOperators(Symbol name) {
```

Comment:
Not a big deal, but I would kind of like to see this returning at::optional. That way it's clear it doesn't raise an lookup failures. This is important here, because the likely usage is
```cpp
for (auto & operator : operators) {
  ...
}
```
and this might silently ignore the fact that there are no operators matched.